src/components/activity/ActivityFormPanel.svelte

```diff
@@ -67,6 +69,12 @@
   $: hasChildren = child_ids ? child_ids.length > 0 : false;
   $: isChild = parent_id !== null;
   $: startTimeField = field<string>(startTime, [required, timestamp]);
+  $: if (duration) {
+    const startTimeISO = new Date(getUnixEpochTime(startTime)).toISOString();
+    endTime = `${getDoyTimeFromDuration(startTimeISO, duration)}`;
```

Comment:
Did you try just passing `startTime` to `getDoyTimeFromDuration`?

Reply:
An example `startTime` is `2020-180T18:02:24.193` but `getDoyTimeFromDuration` expects a valid date string.

Reply:
I think you should be able to do `if (duration && $startTimeField.valid)`

Reply:
Hm but I think the issue here is that JS Date can't parse the DOY format of this startTime.

Reply:
Ahh okay then that makes sense